pinot-broker/src/main/java/com/linkedin/pinot/broker/requesthandler/BrokerRequestHandler.java

```diff
@@ -304,6 +308,19 @@ public BrokerResponse processBrokerRequest(@Nonnull BrokerRequest brokerRequest,
   }
 
   /**
+   * Returns the time column name for the table name from the time boundary service.
+   * Can return null if the time boundary service does not have the information.
+   *
+   * @param tableName Name of table for which to get the time column name
+   * @return Time column name for the table.
+   */
+  @Nullable
+  private String getTimeColumnName(@Nonnull String tableName) {
+    TimeBoundaryInfo timeBoundary = _timeBoundaryService.getTimeBoundaryInfoFor(tableName);
+    return (timeBoundary != null) ? timeBoundary.getTimeColumn() : null;
```

Comment:
Yes, in the PR the optimization will be enabled for hybrid usecases only, and we pick up the time column from offline part of the hybrid usecase.